src/rust/engine/nailgun/src/tests.rs

```diff
@@ -47,8 +46,12 @@ async fn shutdown_awaits_ongoing() {
     let connection_accepted = connection_accepted.clone();
     let should_complete_connection = should_complete_connection.clone();
     move |_| {
-      connection_accepted.notify();
-      Handle::current().block_on(should_complete_connection.notified());
+      connection_accepted.notify_one();
+      loop {
+        if let Some(_) = should_complete_connection.notified().now_or_never() {
+          break;
+        }
+      }
```

Comment:
`block_on` is only available on `Runtime` now and not `Handle`. For now, I have written a busy loop to poll the `Future`, but how we handle blocking work in Tokio 1.0 needs some thought. This busy loop should be changed before the final version of this PR.

cc @stuhood

Reply:
Yea, this blocks the upgrade. See https://github.com/tokio-rs/tokio/issues/2965... have pinged it.

Reply:
`block_on` is now un...blocked! https://github.com/tokio-rs/tokio/issues/3096#issuecomment-803371380

Reply:
Done.